uint8[] index # table of output indices which states are defined
bool[] state
---